File: src/Examples/SimpleWebSite/CS/App_Code/MyItemBase.cs
using N2.Details;

/// <summary>
/// This is an abstract class that we can derive from on in all 
/// situations when we want edit the item's title and name.
/// </summary>
[WithEditable("Name", typeof(N2.Web.UI.WebControls.NameEditor), "Text", 20, "Name")]
public abstract class MyItemBase : N2.ContentItem
{
	[DisplayableLiteral()]
	[EditableTextBox("Title", 10, Required=true)]
	public override string Title
	{
		get { return base.Title; }
		set { base.Title = value; }
	}
}

File: src/Examples/SimpleWebSite/VB/App_Code/MyItemBase.vb
Imports N2.Details

''' <summary>
''' This is an abstract/MustInherit class that we can derive from on in all 
''' situations we want edit the item's title and name.
''' </summary>
''' <remarks></remarks>
<WithEditable("Name", GetType(N2.Web.UI.WebControls.NameEditor), "Text", 20, "Name")> _
Public MustInherit Class MyItemBase
    Inherits N2.ContentItem

    <DisplayableLiteral()> _
    <EditableTextBox("Title", 10)> _
    Public Overrides Property Title() As String
        Get
            Return MyBase.Title
        End Get
        Set(ByVal value As String)
            MyBase.Title = value
        End Set
    End Property

End Class
